File: app/src/main/java/com/grace/placessearch/data/model/Provider.java
package com.grace.placessearch.data.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Provider {

    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("icon")
    @Expose
    private Icon icon;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Icon getIcon() {
        return icon;
    }

    public void setIcon(Icon icon) {
        this.icon = icon;
    }

}

File: app/src/main/kotlin/com/grace/placessearch/common/data/model/Provider.kt
package com.grace.placessearch.common.data.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Provider {

    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("icon")
    @Expose
    var icon: Icon? = null

}